fastplotlib/layouts/_base.py

```diff
@@ -229,6 +228,58 @@ def add_graphic(self, graphic: Graphic, center: bool = True):
             Center the camera on the newly added Graphic
 
         """
+        self._add_or_insert_graphic(graphic=graphic, center=center, action="add")
+
+        graphic.position_z = len(self._graphics)
+
+    def insert_graphic(
+            self,
+            graphic: Graphic,
+            center: bool = True,
+            index: int = 0,
+            z_position: int = None
+    ):
+        """
+        Insert graphic into scene at given position (index) in stored graphics.
+        Replaces graphic at current index if it exists.
+
+        Parameters
+        ----------
+        graphic: Graphic or GraphicCollection
+            Add a Graphic or a GraphicCollection to the plot area at a given position.
+            Note: this must be a real Graphic instance and not a proxy
+
+        center: bool, default True
+            Center the camera on the newly added Graphic
+
+        index: int, default 0
+            Index to insert graphic. Will default replace graphic at index 0. If position is `IndexOutOfBounds`,
+            will place at last possible position.
```

Comment:
This doesn't seem right, shouldn't it be if the passed `index` > len(graphics), then it just appends to the end instead of insert?